src/Host/Broker/Impl/Pipes/ProcessHelpers.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.R.Host.Broker.Pipes {
+    internal static class ProcessHelpers {
```

Comment:
This is not used currently, and `Session` uses `Process` class instead. However, it's likely that this code will need to be revived to avoid storing credentials in memory. Keeping it around as is for now as a prototype for future work.